packages/@aws-cdk/aws-iam/lib/principals.ts

```diff
@@ -0,0 +1,218 @@
+ * Origin Access identities.
+ *
+ * See https://docs.aws.amazon.com/general/latest/gr/acct-identifiers.html
+ *
+ * and
+ *
+ * https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html
+ *
+ * for more details.
+ *
+ */
+export class CanonicalUserPrincipal extends PrincipalBase {
+  constructor(public readonly canonicalUserId: string) {
+    super();
+  }
+
+  public get policyFragment(): PrincipalPolicyFragment {
+    return new PrincipalPolicyFragment({ CanonicalUser: [ this.canonicalUserId ] });
+  }
+}
+
+export class FederatedPrincipal extends PrincipalBase {
+  constructor(
+    public readonly federated: string,
+    public readonly conditions: {[key: string]: any},
+    public assumeRoleAction: string = 'sts:AssumeRole') {
+    super();
+  }
+
+  public get policyFragment(): PrincipalPolicyFragment {
+    return new PrincipalPolicyFragment({ Federated: [ this.federated ] }, this.conditions);
+  }
+}
+
+export class AccountRootPrincipal extends AccountPrincipal {
+  constructor() {
+    super(new StackDependentToken(stack => stack.accountId).toString());
+  }
+}
+
+/**
+ * A principal representing all identities in all accounts
+ */
+export class AnyPrincipal extends ArnPrincipal {
+  constructor() {
+    super('*');
+  }
+}
+
+/**
+ * A principal representing all identities in all accounts
+ * @deprecated use `AnyPrincipal`
+ */
+export class Anyone extends AnyPrincipal { }
+
+export class CompositePrincipal extends PrincipalBase {
+  private readonly principals = new Array<PrincipalBase>();
+
+  constructor(principal: PrincipalBase, ...additionalPrincipals: PrincipalBase[]) {
+    super();
+    this.assumeRoleAction = principal.assumeRoleAction;
+    this.addPrincipals(principal);
+    this.addPrincipals(...additionalPrincipals);
+  }
+
+  public addPrincipals(...principals: PrincipalBase[]): this {
+    for (const p of principals) {
+      if (p.assumeRoleAction !== this.assumeRoleAction) {
+        throw new Error(
+          `Cannot add multiple principals with different "assumeRoleAction". ` +
+          `Expecting "${this.assumeRoleAction}", got "${p.assumeRoleAction}"`);
+      }
+
+      const fragment = p.policyFragment;
+      if (fragment.conditions && Object.keys(fragment.conditions).length > 0) {
+        throw new Error(
+          `Components of a CompositePrincipal must not have conditions. ` +
+          `Tried to add the following fragment: ${JSON.stringify(fragment)}`);
+      }
+
+      this.principals.push(p);
+    }
+
+    return this;
+  }
+
+  public get policyFragment(): PrincipalPolicyFragment {
+    const principalJson: { [key: string]: string[] } = { };
+
+    for (const p of this.principals) {
+      mergePrincipal(principalJson, p.policyFragment.principalJson);
+    }
+
+    return new PrincipalPolicyFragment(principalJson);
+  }
+}
+
+/**
+ * A lazy token that requires an instance of Stack to evaluate
+ */
```

Comment:
Since this is the IAM library, it's not ideal to mix a move and a change. Could you please revert the move so diff will be easier to reason about.